pkg/api/validate/types.go

```diff
@@ -12,6 +12,8 @@ import (
 )
 
 var (
+	rxClusterName = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]{0,61}[a-zA-Z0-9]$`)
```

Comment:
this is almost a rfc1123 hostname portion but 0-9 is missing from the first regex section?  Given it's right next to rxRfc1123 why not actually copy verbatim from below and make this `(?i)^[a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]$` ?

Reply:
this also makes it 1-63 rather than 2-63, fwiw

Reply:
So the reason I decided to not use `rxRfc1123` is that it doesn't permit capital letters, which I thought might be a bit counterintuitive in case customers want to use them in their cluster names. (I believe capital letters should be fine as far as DNS goes?)

The "first portion can't start with 0-9" is copied from the MS suggested regex, and if I had to guess, it's probably an overly cautious legacy parsing thing (e.g. identifiers in C can't start with a digit).

Do we want to allow single character cluster names?

Reply:
`(?i)` means case insensitive in Golang regexps.
RFC952 (precursor to RFC1123) didn't allow initial digits.
Please make it `(?i)^[a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]$` then will merge.

Reply:
Oh, I missed the `(?i)` on the first line.

Had to add some parens to this regex to get it to match properly---otherwise, it would always match the first character if it was alpha-numeric :smile:

Reply:
thanks!